src/main/java/com/gamebros/purepazaak/PurePazaak.java

```diff
@@ -22,22 +42,41 @@ public boolean closeRequested() {
   }
 
   public void render(GameContainer container, Graphics graphics) throws SlickException {
-    graphics.drawString("Pure Pazaak", 50, 50);
-
-    testCard.render(container, graphics);
-    testCard2.render(container, graphics);
+    this.matchController.render(graphics);
+    this.deckController.render(graphics);
+    this.uiController.render(graphics);
   }
 
   public void update(GameContainer container, int delta) {
-    return;
+    this.time += delta;
+
+    // Simulate time thinking.
+    if (this.time >= 1000) {
+      this.ai.update();
+
+      this.time = 0;
+    }
+
+    this.matchController.update();
   }
 
   public void init(GameContainer container) throws SlickException {
-    testCard = new CardComponent(container, new Image("textures/card.png"));
-    testCard.setLocation(300, 300);
+    SideDeckFactory factory = new SideDeckFactory();
+
+    Player playerOne = new Player(factory.build());
+    Player playerTwo = new Player(factory.build());
+
+    Match match = new Match(playerOne, playerTwo);
+
+    this.uiController = new UIController(container, match);
+    this.matchController = new MatchController(match);
+    this.deckController = new DeckController(
+        container,
+        match.getPlayerDeck(PlayerEnum.PLAYERONE),
```

Comment:
`s/getPlayerDeck/getPlayerOneDeck/g`

Reply:
Actually, just pass in `match` to the `DeckController` constructor and resolve the player's deck using `getPlayerOneDeck`.

`public DeckController(GameContainer container, Match match);`